File: Chapter 05/Python/section_5_6_crypto_tracker/before/portfolio_manager.py
import urllib.request
import json

class PortfolioManager:
    """
    THE CORE (Tightly Coupled).
    This class is an architectural liability. It cannot be tested 
    without reaching out into the 'Chaotic Outside World'.
    """

    def calculate_total_value(self, btc_amount: float) -> float:
        """Calculates the total USD value of a Bitcoin balance."""
        
        # VIOLATION 1: Hard-coded network dependency.
        # This ties our business logic directly to the network interface.
        req = urllib.request.Request(
            "https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd",
            headers={"User-Agent": "Python App"}
        )

        try:
            with urllib.request.urlopen(req) as response:
                json_data = response.read()
                
                # VIOLATION 2: The logic is tangled with a specific external JSON format.
                price_data = json.loads(json_data)
                current_price = price_data["bitcoin"]["usd"]
                
                return btc_amount * current_price
        except Exception as e:
            # VIOLATION 3: Unhandled infrastructure exceptions bleeding into domain logic.
            raise RuntimeError(f"Infrastructure failure leaked into Core: {e}")

File: Chapter 05/C#/Section_5_6_Crypto_Tracker/Before/PortfolioManager.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;

namespace Chapter05.CryptoTracker.Before
{
    /// <summary>
    /// THE CORE (Tightly Coupled).
    /// WARNING: This class is a major architectural liability. 
    /// It mixes pure domain logic (calculating portfolio value) with 
    /// infrastructure concerns (making HTTP calls to external APIs).
    /// </summary>
    public class PortfolioManager
    {
        /// <summary>
        /// Calculates the total USD value of a Bitcoin balance.
        /// </summary>
        public decimal CalculateTotalValue(decimal btcAmount)
        {
            // VIOLATION 1: Hard-coded infrastructure dependency.
            // By 'new-ing' up an HttpClient here, we tie our business logic 
            // directly to the network interface.
            using var client = new HttpClient();
            client.DefaultRequestHeaders.Add("User-Agent", "C# App"); 

            // VIOLATION 2: Synchronous execution over the network.
            // If the CoinGecko API goes down, our entire application hangs.
            var json = client.GetStringAsync("https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd").Result;
            
            // VIOLATION 3: The logic is tangled with a specific external JSON format.
            // If CoinGecko changes their response structure, our Core breaks.
            var priceData = JsonSerializer.Deserialize<Dictionary<string, Dictionary<string, decimal>>>(json);
            
            //NULL CHECK
            if (priceData == null || !priceData.ContainsKey("bitcoin"))
            {
                throw new Exception("Failed to parse pricing data from the API.");
            }
            var currentPrice = priceData["bitcoin"]["usd"];
            
            return btcAmount * currentPrice;
        }
    }
}